conseil-common/src/main/scala/tech/cryptonomic/conseil/common/rpc/RpcClient.scala

```diff
@@ -0,0 +1,75 @@
+package tech.cryptonomic.conseil.common.rpc
+
+import scala.util.control.NoStackTrace
+
+import cats.effect.Concurrent
+import com.typesafe.scalalogging.LazyLogging
+import fs2.Stream
+import org.http4s.client._
+import org.http4s.client.dsl.Http4sClientDsl
+import org.http4s.{EntityDecoder, EntityEncoder, Method, Uri, Header}
+
+import tech.cryptonomic.conseil.common.rpc.RpcClient._
+
+class RpcClient[F[_]: Concurrent](
+    endpoint: String,
+    maxConcurrent: Int,
+    client: Client[F],
+    headers: Header*
+) extends Http4sClientDsl[F]
+    with LazyLogging {
+
+  def stream[P, R](batchSize: Int)(request: Stream[F, RpcRequest[P]])(
+      implicit decode: EntityDecoder[F, Seq[RpcResponse[R]]],
+      encode: EntityEncoder[F, List[RpcRequest[P]]]
+  ): Stream[F, R] =
+    request
+      .chunkN(batchSize)
+      .evalTap { requests =>
+        Concurrent[F].delay(logger.info(s"Call RPC in a batch of: ${requests.size}"))
+
+      }
+      .mapAsyncUnordered(maxConcurrent) { chunks =>
+        client
+          .expect[Seq[RpcResponse[R]]](
+            Method.POST(chunks.toList, Uri.unsafeFromString(endpoint), headers: _*)
```

Comment:
I'm missing something here, please: is the endpoint always the same?

Reply:
yeap, In JSON-RPC is only one POST endpoint

Reply:
Thanks, I see I should have a look at json-rpc